BlockchainSdk/Blockchains/Tron/TronWalletManager.swift

```diff
@@ -262,38 +262,71 @@ extension TronWalletManager: TronTransactionDataBuilder {
+    func sendStakeKit(transactions: [StakeKitTransaction], signer: TransactionSigner) -> AnyPublisher<[TransactionSendResult], SendTxError> {
+
+        let presignedInputsPublisher = Result {
+            try transactions.map {
+                try TronStakeKitTransactionHelper().prepareForSign($0.unsignedData)
+            }
+        }.publisher
+
+        let readyToSendTransactionsPublisher = presignedInputsPublisher
+            .withWeakCaptureOf(self)
+            .flatMap { manager, inputs -> AnyPublisher<[Data], Error> in
+                signer
+                    .sign(hashes: inputs.map { $0.hash }, walletPublicKey: manager.wallet.publicKey)
+                    .tryMap { signatures -> [Data] in
+                        try zip(inputs, signatures).map { (input, signature) in
+                            try manager.txBuilder.buildForSend(rawData: input.rawData, signature: signature)
+                        }
+                    }
+                    .eraseToAnyPublisher()
+            }
+
+        let sentTransactionsPublisher = readyToSendTransactionsPublisher
+            .flatMap { [weak self] transactionsData -> AnyPublisher<[TronBroadcastResponse], Error> in
+                guard let self else {
+                    return .anyFail(error: WalletError.empty)
                 }
-        }
-        .withWeakCaptureOf(self)
-        .flatMap { manager, data in
-            manager.networkService
-                .broadcastHex(data)
-                .mapSendError(tx: data.hexString)
-        }
-        .withWeakCaptureOf(self)
-        .tryMap { manager, broadcastResponse in
-            guard broadcastResponse.result == true else {
-                throw WalletError.failedToSendTx
+
+                let sendPublishers = transactionsData
+                    .map { data in
+                        self.networkService
+                            .broadcastHex(data)
+                            .mapSendError(tx: data.hexString)
+                            .eraseToAnyPublisher()
+                    }
+
+                return Publishers.Sequence(sequence: sendPublishers)
+                    .flatMap(maxPublishers: .max(1)) { $0 }
+                    .collect()
+                    .eraseToAnyPublisher()
             }
 
-            let hash = broadcastResponse.txid
-            let mapper = PendingTransactionRecordMapper()
-            let record = mapper.mapToPendingTransactionRecord(stakeKitTransaction: transaction, hash: hash)
-            manager.wallet.addPendingTransaction(record)
-            return TransactionSendResult(hash: hash)
-        }
-        .eraseSendError()
-        .eraseToAnyPublisher()
+        return sentTransactionsPublisher
+            .withWeakCaptureOf(self)
+            .tryMap { manager, broadcastResponses -> [TransactionSendResult] in
+                guard broadcastResponses.allSatisfy({ $0.result == true }) else {
+                    throw WalletError.failedToSendTx
```

Comment:
можно так
```suggestion
                guard broadcastResponses.allSatisfy({ $0.result }) else {
                    throw WalletError.failedToSendTx
```

А вообще `allSatisfy` хитрый, и если `broadcastResponses.isEmpty == true` то но тоже будет `true`
можно как вариант добавить )